src/clp_ffi_py/encoder/encoding_methods.cpp

```diff
@@ -0,0 +1,138 @@
+#include <clp_ffi_py/ErrorMessage.hpp>
+#include <clp_ffi_py/Python.hpp>
+#include <clp_ffi_py/encoder/encoding_methods.hpp>
+
+#include <clp/components/core/src/ffi/encoding_methods.hpp>
+#include <clp/components/core/src/ffi/ir_stream/encoding_methods.hpp>
+#include <clp/components/core/src/type_utils.hpp>
+
+// NOLINTBEGIN(cppcoreguidelines-init-variables)
+// NOLINTBEGIN(cppcoreguidelines-pro-type-vararg)
+
+namespace clp_ffi_py::encoder::four_byte_encoding {
+auto encode_preamble(PyObject* Py_UNUSED(self), PyObject* args) -> PyObject* {
+    ffi::epoch_time_ms_t ref_timestamp;
+    char const* input_timestamp_format;
+    char const* input_timezone;
+    Py_ssize_t input_timestamp_format_size;
+    Py_ssize_t input_timezone_size;
+
+    if (0 == PyArg_ParseTuple(
+                     args,
+                     "Ls#s#",
+                     &ref_timestamp,
+                     &input_timestamp_format,
+                     &input_timestamp_format_size,
+                     &input_timezone,
+                     &input_timezone_size)) {
+        return nullptr;
+    }
+
+    std::string const timestamp_format{
+            input_timestamp_format,
+            static_cast<size_t>(input_timestamp_format_size)};
+    std::string const timezone{input_timezone, static_cast<size_t>(input_timezone_size)};
+    std::string const timestamp_pattern_syntax{};
```

Comment:
The curly braces are unnecessary/inconsistent here (`std::string` has a default constructor) - compare to the next line declaring a `std::vector<int8_t> buf` and line 63 declaring `std::string logtype` (without curly braces)

Reply:
Unlike buf and logtype, this variable is actually an input argument but it's not yet used by any of IR features yet. I'm using {} to explicitly tell "this is empty". If it looks confusing I will add a comment or leave the work to the default constructor.